### ISSUE TITLE
Extracting text nodes from a specific element

### ISSUE THREAD

**Issue by Xliff** [MARTIMM/XmlActions#3]
Given the following XML snippet:

```
<stuff>
  This should be valid...
  <foo bar="ber">Baz</foo>
  <extra>More text in stuff.</extra>
  ...XML data, right?
</stuff>
```

What would be the best way to organize an action class so that you can pull the "Baz" from the <foo> tag, and still pull the "This should be valid... Baz More text in stuff...XML data, right?" from the <stuff> tag?

I can think of a few ways to do it, but I am always running into the problem of inner text nodes overwriting the outer text nodes.

**Comment by MARTIMM** [comment]
@Xliff 

I am not sure what you really want. Is it that you need to remove an element or do you want to copy an element and place it somewhere else? The XML package is always replacing parent links in the element node when you pull the element out or move it elsewhere. This is the reason that elements or text can disappear. You might want to clone it or make a fresh copy of the element. You can look up the documentation of XML to get the proper actions.

To skip through the text you can do this;

```
  method xml:text ( Array $parent-path, Str $text is copy ) {
    # check if there is on spaces
    next if $text ~~ m/^ \s* $/;

    # remove leading and trailing spaces
    $text ~~ s/^ \s+ //;
    $text ~~ s/ \s+ $//;

    # then do someting with it depending on parent element
    note 'PP: ', $parent-path[*-1].name, ', ', $text;
  }
```

**Comment by Xliff** [comment]
@MARTIMM 

Ah, no. I am not to replace text, just to extract it at a _node_ level. Consider my previous example:

```
<stuff>
  This should be valid...
  <foo bar="ber">Baz</foo>
  <extra>More text in stuff.</extra>
  ...XML data, right?
</stuff>
```

Is there a way to get the text node from the &lt;foo&gt; &lt;extra&gt; and &lt;stuff&gt; elements without going through xml:text()

**Comment by MARTIMM** [comment]
No, I'm sorry, there is no method in this package to find text in another way. Perhaps when you know that there is text at some element try the following using the methods from XML
```
  method stuff ( Array $parent-path  ) {
     for $parent-path[*-1].contents -> $txt { say "'$txt'"; }
  }
```
This gives a list of text elements found under that tag still with some spaces around it.
